### PR TITLE
added mixed precision, disabled gradient tracking and added tqdm bar

### PR DESCRIPTION
**Pull request by bluepeach1121**
- I wrapped inference in torch.no_grad() + torch.autocast for mixed precision. it helps to speed up the execution
- Replace torch.utils.benchmark.Timer with a tqdm loop, allows us to track the progress better

### CODE REVIEW

**Comment by bluepeach1121** [comment]
Please check and verify

**Comment by lusxvr** [comment]
The `torch.no_grad()` is already applied in the `generate()` method of the model!

Additionally, the mixed precision gives us quicker inference, but unfortunately also reduces the compatibility with different setups.

File: benchmark-inference.py [hunk 1]
```diff
@@ -8,36 +10,40 @@
...
 
 if __name__ == "__main__":
     model = VisionLanguageModel.from_pretrained("lusxvr/nanoVLM-222M").to(device)
     model.eval()
-    
+
     tokenizer = get_tokenizer(model.cfg.lm_tokenizer)
     image_processor = get_image_processor(model.cfg.vit_img_size)
 
+    # 5) Prepare inputs
```
Comment (andimarafioti) [inline benchmark-inference.py:28]: Why is this #5) 😭 LLMs should be smarter in May 2025 😅

Comment (bluepeach1121) [reply benchmark-inference.py:28]: 😂 i didnt notice that

**Comment by bluepeach1121** [comment]
okay, can I remove all of that and leave the tqdm wrapper? Then I update the pull request?

**Comment by andimarafioti** [comment]
Hi! We prefer staying with the pure pytorch implementation here. But thanks for your interest in the project, we have a few suggestions for contributions if you're looking for low hanging fruits :)
